packages/fxa-auth-server/lib/routes/password.ts

```diff
@@ -2,65 +2,72 @@
+import { schema as METRICS_CONTEXT_SCHEMA } from '../metrics/context';
+import { gleanMetrics } from '../metrics/glean';
+import * as requestHelper from '../routes/utils/request_helper';
+import { AuthLogger, AuthRequest } from '../types';
+import { recordSecurityEvent } from './utils/security-event';
+import * as validators from './validators';
 
-const validators = require('./validators');
 const HEX_STRING = validators.HEX_STRING;
 
-const butil = require('../crypto/butil');
-const error = require('../error');
-const isA = require('joi');
-const random = require('../crypto/random');
-const requestHelper = require('../routes/utils/request_helper');
-const { recordSecurityEvent } = require('./utils/security-event');
-const { emailsMatch } = require('fxa-shared').email.helpers;
-const { OtpManager } = require('@fxa/shared/otp');
+class OtpRedisAdapter implements OtpStorage {
+  constructor(private redis: Redis, private ttl: number) {}
 
-const PASSWORD_DOCS = require('../../docs/swagger/password-api').default;
-const DESCRIPTION = require('../../docs/swagger/shared/descriptions').default;
-const METRICS_CONTEXT_SCHEMA = require('../metrics/context').schema;
+  async set(key: string, value: string) {
+    await this.redis.set(key, value, 'EX', this.ttl);
+    return null;
+  }
+
+  async get(key: string) {
+    return this.redis.get(key);
+  }
+
+  async del(key: string) {
+    await this.redis.del(key);
+    return null;
+  }
+}
 
 module.exports = function (
-  log,
-  db,
-  Password,
-  redirectDomain,
-  mailer,
-  verifierVersion,
-  customs,
-  signinUtils,
-  push,
-  config,
-  oauth,
-  glean,
-  authServerCacheRedis,
-  statsd
+  log: AuthLogger,
+  db: any,
+  Password: any,
+  redirectDomain: any,
+  mailer: any,
+  verifierVersion: any,
+  customs: any,
+  signinUtils: any,
+  push: any,
+  config: ConfigType,
+  oauth: any,
+  glean: ReturnType<typeof gleanMetrics>,
+  authServerCacheRedis: Redis,
+  statsd: StatsD
 ) {
   const otpUtils = require('../../lib/routes/utils/otp')(log, config, db);
   const otpRedisAdapter = config.passwordForgotOtp.enabled
-    ? {
-        set: async (key, val) => {
-          return authServerCacheRedis.set(
-            key,
-            val,
-            'EX',
-            config.passwordForgotOtp.ttl
-          );
-        },
-        get: async (key) => {
-          return authServerCacheRedis.get(key);
-        },
-        del: async (key) => {
-          return authServerCacheRedis.del(key);
-        },
-      }
-    : { set: () => {}, get: () => {}, del: () => {} };
+    ? new OtpRedisAdapter(authServerCacheRedis, config.passwordForgotOtp.ttl)
+    : { set: async () => null, get: async () => null, del: async () => null };
   const otpManager = new OtpManager(
     { kind: 'passwordForgot', digits: config.passwordForgotOtp.digits },
     otpRedisAdapter
   );
 
-  function failVerifyAttempt(passwordForgotToken) {
+  function failVerifyAttempt(passwordForgotToken: any) {
```

Comment:
Nit: the required type needed for the calls below is:
`{ id:string, uid:string, tries:number }`

Again, maybe a good follow up for adding in more types.